service/storage/operations_internal_test.go

```diff
@@ -1324,6 +1324,34 @@ func TestLibrary_IterateLedger(t *testing.T) {
 		assert.Equal(t, payloads[entries-1], got[path])
 	})
 
+	t.Run("handles exclusion of payloads below specified height", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		codec := zbor.NewCodec()
+		l := &Library{codec}
+
+		for i := 0; i < entries; i++ {
+			height := mocks.GenericHeight + uint64(i)
+			require.NoError(t, db.Update(l.SavePayload(height, paths[i], payloads[i])))
+		}
```

Comment:
I would suggest doing the DB insertions right after declaring the DB instead:

```suggestion
		db := helpers.InMemoryDB(t)
		defer db.Close()

		for i := 0; i < entries; i++ {
			height := mocks.GenericHeight + uint64(i)
			require.NoError(t, db.Update(l.SavePayload(height, paths[i], payloads[i])))
		}

		codec := zbor.NewCodec()
		l := &Library{codec}
```

Reply:
The DB updates rely on codec/library, so they need to come first.

Reply:
Then the following:

```suggestion
		codec := zbor.NewCodec()
		l := &Library{codec}
		
		db := helpers.InMemoryDB(t)
		defer db.Close()

		for i := 0; i < entries; i++ {
			height := mocks.GenericHeight + uint64(i)
			require.NoError(t, db.Update(l.SavePayload(height, paths[i], payloads[i])))
		}
```

But everywhere else, the DB is created and then filled in two subsequent blocks, so I'd like to keep it consistent. Also it makes more sense for me when reading the code, that the DB itself is the more immediate dependency of this block, rather than the library and the codec.

Reply:
I'm not sure if I'm reading something wrong. But it's the other way around - it is consistent now, since all of the other test cases in `TestLibrary_IterateLedger` first initialize the db, then the codec/library, then populate the db.

I don't really have a big preference since it's pretty trivial, but it's currently in line with the existing code. I could just switch it everywhere though, if you think it's neater.

Reply:
Then there's already an inconsistency in this file indeed :p Because I'm pretty sure everywhere else the DB comes right before it's filled.

I do think it's neater indeed 😄